### ISSUE TITLE
The "Device credentials" feature doesn't work with fingerprints

### ISSUE THREAD

**Issue by gradinaruvasile** [nextcloud/android#3026]
### Actual behaviour
- If the "Device credentials" lock feature is selectedif a device has both passphrase and fingerprint set to unlock, only the passphrase is working to unlock the app.

### Expected behaviour
- If the "Device credentials" lock feature is selectedif a device has both passphrase and fingerprint set to unlock, either of these should unlock the app.
 
### Steps to reproduce
1. Set up both encryption passphrase and fingerprint for unlocking
2. Set "Device credentials" in the NC app
3. Close then open the app and only the passphrase unlocks the app

Although the screen that requests the credentials has the fingerprint icon below the passphrase input area and if the sensor is touched this icon turns green but the password is asked again.
BTW for example AndOTP has this same feature and it works correctly with fingerprint too.

### Environment data
Android version:
8.1
Device model: 
Lenovo K6 power
Stock or customized system:
Lineage OS
Nextcloud app version:
3.3.0 RC2
Nextcloud server version:
14.0
### Logs

```
09-12 14:06:49.128  2185 24085 I ActivityManager: START u0 {flg=0x2000000 cmp=com.android.settings/.password.ConfirmLockPassword (has extras)} from uid 1000
09-12 14:06:49.325  1925  1925 D fpc_fingerprint_hal: fpc_get_authenticator_id
09-12 14:06:49.325  1925  1925 I fpc_fingerprint_hal: fpc_worker_set_state_locked worker_state_t = 0
09-12 14:06:49.328  1925  1925 D fpc_fingerprint_hal: fpc_get_authenticator_id id 10071650917609000870
09-12 14:06:49.328  1925  1925 I fpc_fingerprint_hal: fpc_worker_set_state_locked worker_state_t = -1
09-12 14:06:49.328  1925  1925 D fpc_fingerprint_hal: fpc_get_authenticator_id
09-12 14:06:49.328  1925  1925 I fpc_fingerprint_hal: fpc_worker_set_state_locked worker_state_t = 0
09-12 14:06:49.329  1925  2066 I fpc_fingerprint_hal: workerFunction STATE_NONE
09-12 14:06:49.331  1925  1925 D fpc_fingerprint_hal: fpc_get_authenticator_id id 10071650917609000870
09-12 14:06:49.331  1925  1925 I fpc_fingerprint_hal: fpc_worker_set_state_locked worker_state_t = -1
09-12 14:06:49.335 11162 28263 W TileUtils: Found com.android.settings.backup.BackupSettingsActivity for intent Intent { act=com.android.settings.action.SETTINGS pkg=com.android.settings } missing metadata com.android.settings.category
09-12 14:06:49.338  1925  2066 I fpc_fingerprint_hal: workerFunction STATE_IDLE
09-12 14:06:49.211  2185 24085 I ActivityManager: START u0 {flg=0x2000000 cmp=com.android.settings/.password.ConfirmLockPassword (has extras)} from uid 1000
09-12 14:06:49.338  2185  2185 V FingerprintService: startAuthentication(com.android.settings)
09-12 14:06:49.338  2185  2185 V FingerprintService: starting client AuthenticationClient(com.android.settings), initiatedByClient = true)
09-12 14:06:49.338  1925  1925 D fpc_fingerprint_hal: fpc_authenticate operation_id 0
09-12 14:06:49.338  1925  1925 I fpc_fingerprint_hal: fpc_worker_set_state_locked worker_state_t = 0
09-12 14:06:49.338  1925  1925 I fpc_fingerprint_hal: fpc_worker_set_state_locked worker_state_t = 2
09-12 14:06:49.339  1925  2066 D fpc_fingerprint_hal: workerFunction STATE_AUTHENTICATE
09-12 14:06:49.339  2185  2185 W FingerprintService: client com.android.settings is authenticating...
09-12 14:06:49.340 11162 28263 W TileUtils: Found org.lineageos.lineageparts.trust.TrustPreferences for intent Intent { act=org.lineageos.lineageparts.action.SETTINGS pkg=org.lineageos.lineageparts } missing metadata 
09-12 14:06:49.346  1925  2066 I fpc_tac : fpc_tac_begin_identify begin
09-12 14:06:49.362  1925  2066 I fpc_tac : fpc_tac_begin_identify end
09-12 14:06:49.390 11162 28264 D Settings: No enabled state changed, skipping updateCategory call
09-12 14:06:49.506  2185  2270 I ActivityManager: Displayed com.android.settings/.password.ConfirmLockPassword: +204ms (total +472ms)
09-12 14:06:49.676 15138 15138 D RequestCredentialsActivity: onSaveInstanceState(Bundle) starting
09-12 14:06:49.677 15138 15138 D RequestCredentialsActivity: onStop() ending
09-12 14:06:49.825 28378 28378 W tfc     : Failed to find local clusters.
09-12 14:06:50.084 28378 28378 I zygote64: Deoptimizing void ukg.a(usl, uso) due to JIT inline cache
09-12 14:06:50.377 28378 28378 W uza     : sections == null.  missed changes on list uba{type=SAPI_LL_TYPE_SECTIONED_INBOX_PRIMARY_UNREAD, priority=DEFAULT, pauseState=RESUMED, #elements=7}
09-12 14:06:50.876  2185  2185 W WindowManager: removeWindowToken: Attempted to remove non-existing token: android.os.Binder@bb0ac2d
09-12 14:06:50.381 28378 28378 W uza     : sections == null.  missed changes on list uba{type=SAPI_LL_TYPE_SECTIONED_INBOX_PRIMARY_UNREAD, priority=DEFAULT, pauseState=RESUMED, #elements=7}

```

#### Web server error log
```
Insert your webserver log here
```

#### Nextcloud log (data/nextcloud.log)
```
Insert your Nextcloud log here
```
**NOTE:** Be super sure to remove sensitive data like passwords, note that everybody can look here! You can use the Issue Template application to prefill some of the required information: https://apps.nextcloud.com/apps/issuetemplate

**Comment by joekerna** [comment]
This might be a duplicate of #2879

**Comment by gradinaruvasile** [comment]
This seems to be fixed in app version 3.3.2. Now the device credentials can unlock the app with fingerprint too.

**Comment by AndyScherzinger** [comment]
Thank you for your feedback 👍
